Makefile

```diff
@@ -8,27 +8,44 @@ endif
 
 .PHONY: dev
 dev: ## Installs dbt-* packages in develop mode along with development dependencies.
-	pip install -r dev-requirements.txt -r editable-requirements.txt
+	@\
+	pip install -r dev-requirements.txt -r editable-requirements.txt && \
+	pre-commit install
 
 .PHONY: mypy
 mypy: .env ## Runs mypy for static type checking.
-	$(DOCKER_CMD) tox -e mypy
+	@\
+	$(DOCKER_CMD) pre-commit run mypy | grep -v "[INFO]"
 
 .PHONY: flake8
 flake8: .env ## Runs flake8 to enforce style guide.
-	$(DOCKER_CMD) tox -e flake8
+	@\
+	$(DOCKER_CMD) pre-commit run flake8 | grep -v "[INFO]"
+
+.PHONY: black
+black: .env ## Runs black to enforce style guide.
+	@\
+	$(DOCKER_CMD) pre-commit run black | grep -v "[INFO]"
 
 .PHONY: lint
 lint: .env ## Runs all code checks in parallel.
-	$(DOCKER_CMD) tox -p -e flake8,mypy
+	@\
+	$(DOCKER_CMD) pre-commit run flake8 | grep -v "[INFO]"; \
+	$(DOCKER_CMD) pre-commit run mypy | grep -v "[INFO]"; \
+	$(DOCKER_CMD) pre-commit run black | grep -v "[INFO]" \
```

Comment:
This will only run on git staged files. Is that what is intended here?

Reply:
oh I def want it to run on everything. I never stage before running `make lint`

Reply:
It sounds like we have some different use cases for how we use `make lint`.

A: Yes black is now included in the linter step and it will make changes.  I get how that's not ideal so I can change it to just output the errors/warnings without making changes.

B: I had imagined using it just for changes.. but it that's not the way we want it to go I'm happy to change it to run on everything.

Reply:
maybe keep `make lint` as close to the same as it was, but have a new command `make format` or `make pre-commit` or whatever that does all the things on staged files? 🤷‍♂️

Reply:
TBH, I never use `make lint` and no-one else has said anything, so I think you get to pick how you want it to work @nathaniel-may

Reply:
`make lint`: runs mypy and flake8 in parallel on all source files but makes no changes to them.

Reply:
There's a fair bit going on in terms of ways we use this so here's a matrix to describe the current state.
|                    command                   |                                                            hooks run                                                            | files checked  | writes code changes |
|:--------------------------------------------:|:-------------------------------------------------------------------------------------------------------------------------------:|----------------|---------------------|
| `make dev`*                                  | -                                                                                                                               | -              | -                   |
| `make black`<br>`make flake8`<br>`make mypy` | linter/formatter <br>(as specified)                                                                                             | staged changes | no                  |
| `make lint`                                  | black, flake8, mypy                                                                                                             | staged changes | no                  |
| `make test`                                  | black, flake8, mypy                                                                                                             | staged changes | no                  |
| `git commit`                                 | check-yaml, check-json,<br>fix-end-of-files, <br>trim-trailing-whitespace,<br>check-for-case-conflicts, <br>black, flake8, mypy | staged changes | yes**               |
| `pre-commit run --all-files`                 | check-yaml, check-json,<br>fix-end-of-files, <br>trim-trailing-whitespace,<br>check-for-case-conflicts, <br>black, flake8, mypy | all files      | yes**               |


* installs pre-commit hooks, required before any of this will work.
** except `mypy` and `flake8`, which don't support that feature.

Reply:
Finding a way to run this on unstaged changes without running it on everything in the entire repo is proving a bit challenging so I'm going to put that in as a tech debt ticket.

Reply:
Removed black from `make lint` and updated the matrix above.